src/v/cloud_storage/partition_manifest.h

```diff
@@ -98,6 +98,38 @@ class partition_manifest final : public base_manifest {
     /// Create manifest for specific ntp
     explicit partition_manifest(model::ntp ntp, model::initial_revision_id rev);
 
+    template<class segment_t>
+    partition_manifest(
+      model::ntp ntp,
+      model::initial_revision_id rev,
+      model::offset so,
+      model::offset lo,
+      const std::vector<segment_t>& segments,
+      const std::vector<segment_t>& replaced)
+      : _ntp(std::move(ntp))
+      , _rev(rev)
+      , _last_offset(lo)
+      , _start_offset(so) {
+        for (const auto& nm : replaced) {
+            auto key = parse_segment_name(nm.name);
+            vassert(
+              key.has_value(),
+              "can't parse name of the replaced segment in the manifest '{}'",
```

Comment:
Any particular reason this parse check is a vassert but the one a few lines down is a runtime_error?